editor/components/block-settings-menu/index.js

```diff
@@ -94,13 +98,12 @@ export class BlockSettingsMenu extends Component {
 					renderContent={ ( { onClose } ) => (
 						// Should this just use a DropdownMenu instead of a DropDown ?
 						<NavigableMenu className="editor-block-settings-menu__content">
-							{ renderBlockMenu( { onClose, children: [
-								count === 1 && <BlockModeToggle key="mode-toggle" uid={ firstBlockUID } onToggle={ onClose } role="menuitem" />,
-								count === 1 && <UnknownConverter key="unknown-converter" uid={ firstBlockUID } role="menuitem" />,
-								<BlockDuplicateButton key="duplicate" uids={ uids } rootUID={ rootUID } role="menuitem" />,
-								count === 1 && <SharedBlockSettings key="shared-block" uid={ firstBlockUID } onToggle={ onClose } itemsRole="menuitem" />,
-								<BlockTransformations key="transformations" uids={ uids } onClick={ onClose } itemsRole="menuitem" />,
-							] } ) }
+							<PluginsBlockSettingsMenuSlot fillProps={ { onClose } } />
```

Comment:
Questions:

 - This adds the menu items to the top of the list, should we add another slot after the menu items?
 - We don't use any prop (aside onClose) in our current usage, but I guess if we want to expose this to plugins, we'd probably need `uids`?

Thoughts?

Reply:
> This adds the menu items to the top of the list, should we add another slot after the menu items?

I don't know what is the exact use case in here, but I don't feel very good about having two slots only to have a way to pick whether we should prepend or append menu items. I'm wondering if we shouldn't rethink the way all slot operate. There are a few things we can explore. First of all, we could convert all menu items into fills to have better controls over them and:
* Introduce priority prop for fills.
* Add a filter for a slot which would allow reordering or modifications of fills.

...or maybe use a filter instead of Slot/Fill in the first place.

In general, there is an expectation from plugin developers to have a way to remove some of the fills that are rendered. We can provide an API for that, or find a way to guard fills with the check which allows preventing their rendering if the given condition is met.

> We don't use any prop (aside `onClose`) in our current usage, but I guess if we want to expose this to plugins, we'd probably need `uids`?

`count` might make sense, too.

Reply:
Yes, I agree it's not perfect, Slots are limited to specific positions. A `priority/order` support makes sense. That said, I think it's a bigger problem than what this PR is trying to solve and it concerns all the other slots.

Like you said, I think we should move on like this, for now, it's way better than the  `renderBlockMenu` prop and address the slots flexibility in a dedicated issue/PR.